### ISSUE TITLE
No apps open when tapped

### ISSUE THREAD

**Issue by luskin** [includable/react-native-map-link#25]
We installed the package and the UI for the popup looks fantastic when opened, however when we click on any of the mapping options (apple/google/uber/lyft/waze) nothing happens, the modal just disappears and the following warning is generated in the console:

```
Possible Unhandled Promise Rejection (id: 1):
TypeError: Cannot read property 'indexOf' of undefined
TypeError: Cannot read property 'indexOf' of undefined
    at checkOptions (blob:file:///722dfb5c-3809-4aaa-bd85-0cea3ed24ad4:110095:59)
    at showLocation$ (blob:file:///722dfb5c-3809-4aaa-bd85-0cea3ed24ad4:106794:37)
    at tryCatch (blob:file:///722dfb5c-3809-4aaa-bd85-0cea3ed24ad4:18420:19)
    at Generator.invoke [as _invoke] (blob:file:///722dfb5c-3809-4aaa-bd85-0cea3ed24ad4:18593:24)
    at Generator.prototype.(anonymous function) [as next] (blob:file:///722dfb5c-3809-4aaa-bd85-0cea3ed24ad4:18463:23)
    at tryCatch (blob:file:///722dfb5c-3809-4aaa-bd85-0cea3ed24ad4:18420:19)
    at invoke (blob:file:///722dfb5c-3809-4aaa-bd85-0cea3ed24ad4:18496:22)
    at blob:file:///722dfb5c-3809-4aaa-bd85-0cea3ed24ad4:18524:13
    at tryCallTwo (blob:file:///722dfb5c-3809-4aaa-bd85-0cea3ed24ad4:5193:7)
    at doResolve (blob:file:///722dfb5c-3809-4aaa-bd85-0cea3ed24ad4:5357:15)
```

Are there any additional setup steps not included in the README such as linking or anything? I put the required fields in on the Info.plist but there seems to be no other installation steps. Thanks, looks like a great package!

Our implementation is as follows:

```
        <Popup
          isVisible={this.state.navigationDirectionsPopupVisible}
          onCancelPressed={() => this.setState({ navigationDirectionsPopupVisible: false })}
          onAppPressed={() => this.setState({ navigationDirectionsPopupVisible: false })}
          onBackButtonPressed={() => this.setState({ navigationDirectionsPopupVisible: false })}
          modalProps={{
            // you can put all react-native-modal props inside.
            animationIn: 'slideInUp'
          }}
          options={{
            latitude: 38.8976763,
            longitude: -77.0387185,
            title: 'The White House', // optional
            dialogTitle: 'Get Directions', // optional (default: 'Open in Maps')
            dialogMessage: 'Choose your prefered application', // optional (default: 'What app would you like to use?')
            cancelText: 'Nevermind' // optional (default: 'Cancel')
          }}
          style={
            {
              /* Optional: you can override default style by passing your values. */
            }
          }
        />
```

Additionally the `showLocation` method does nothing. When we call it no actionsheet pops up like the documentation states it will.

```
OS: iOS 11
Device: iPhone X
React: v0.55
```

**UPDATE:**
I dug through your code and noticed that upon tapping the following is called:
```
  _onAppPressed = ({app}) => {
    showLocation({...this.props.options, app})
    this.props.onAppPressed()
```

This makes sense that nothing is happening since the `showLocation` does not work for me directly either. The problem lies within `showLocation`

**UPDATE 2:**
Diving into your code it looks like it's an error in your `utils.js` file on line 121 `apps.indexOf(options.app) < 0` - I have confirmed that for whatever reason apps is undefined so it silently fails and nothing happens. Please advise where we should go from here. If I wrap this in a try/catch the mapping applications open fine but curious why this is happening.

**UPDATE 3:**
Ok, it's importing `{ apps }` from `constants/index.js` however there is no `apps` exported from constants/index.js.... so looks just like coder error and a lack of tests. I have forked to my own repo for fix.

**Comment by johanleroch** [comment]
We saw this but didn't have to time to fix it at the moment, I'll try this we. 
Meanwhile you can use the popup on my branch: https://github.com/johanleroch/react-native-map-link/tree/styling that was the code I made before the PR has been merged.

`yarn add johanleroch/react-native-map-link#styling`
or 
`npm install --save johanleroch/react-native-map-link#styling`

Sorry for the inconvenience and the delay.
Regards

**Comment by tschoffelen** [comment]
@luskin Sorry, we did indeed merge that new popup rather quickly, due to lack of time on my part to check it... 

Would you mind opening a PR with your fix? Would appreciate it very much!

**Comment by tschoffelen** [comment]
@luskin Once again sorry about this, and @johanleroch thanks for responding so quickly. I've just released 2.0.3 which fixes this.
